gadgets/fsnotify/test/unit/fsnotify_test.go

```diff
@@ -0,0 +1,230 @@
+
+						FaType:     utils.NormalizedStr,
+						FaResponse: utils.NormalizedStr,
+
+						IWd:     utils.NormalizedInt,
+						IIno:    eventDetails.Ino,
+						IInoDir: eventDetails.InoDir,
+
+						Name:  eventDetails.FileName,
+					}
+				})(t, info, 0, events)
+			},
+		},
+	}
+	for name, testCase := range testCases {
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			var eventDetails EventDetails
+			runner := utilstest.NewRunnerWithTest(t, testCase.runnerConfig)
+
+			normalizeEvent := func(event *ExpectedFsnotifyEvent) {
+				utils.NormalizeString(&event.Timestamp)
+				utils.NormalizeInt(&event.TracerMntnsId)
+
+				utils.NormalizeString(&event.FaType)
+				utils.NormalizeString(&event.FaResponse)
+			}
+			onGadgetRun := func(gadgetCtx operators.GadgetContext) error {
+				utilstest.RunWithRunner(t, runner, func() error {
+					var err error
+					eventDetails, err = testCase.generateEvent()
+					if err != nil {
+						return err
+					}
+					return nil
+				})
+				return nil
+			}
+			opts := gadgetrunner.GadgetRunnerOpts[ExpectedFsnotifyEvent]{
+				Image:          "fsnotify",
+				Timeout:        10 * time.Second,
+				OnGadgetRun:    onGadgetRun,
+				NormalizeEvent: normalizeEvent,
+			}
+
+			gadgetRunner := gadgetrunner.NewGadgetRunner(t, opts)
+
+			gadgetRunner.RunGadget()
+
+			testCase.validateEvent(t, runner.Info, eventDetails, gadgetRunner.CapturedEvents)
+		})
+	}
+}
+
+func generateEvent() (EventDetails, error) {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return EventDetails{}, err
+	}
+	defer watcher.Close()
+
+	err = watcher.Add(os.TempDir())
+	if err != nil {
+		return EventDetails{}, err
+	}
+
+	// event 1
+	newFile, err := os.CreateTemp(os.TempDir(), "test-*.txt")
+	if err != nil {
+		return EventDetails{}, err
+	}
+	defer newFile.Close()
+
+	// event 2
+	_, err = newFile.WriteString("Hello, fsnotify!")
+	if err != nil {
+		return EventDetails{}, err
+	}
+
+	inode, dirInode, err := calculateInodeValues(newFile.Name())
+	if err != nil {
+		return EventDetails{}, err
+	}
+
+	fileName := path.Base(newFile.Name())
+	eventDetails := EventDetails{
+		FileName: fileName,
+		Ino:      uint32(inode),
+		InoDir:   uint32(dirInode),
+	}
+	return eventDetails, nil
+}
+
+func calculateInodeValues(fileName string) (uint64, uint64, error) {
+	// extract inode info about file
+	fileInfo, err := os.Stat(fileName)
+	if err != nil {
+		return 0, 0, err
+	}
+	fileSys := fileInfo.Sys()
+	var inode uint64
+	if stat, ok := fileSys.(*syscall.Stat_t); ok {
+		inode = uint64(stat.Ino)
+	}
+	fmt.Printf("Inode of File: %d\n", inode)
+
+	// extract inode info about directory
+	dirInfo, err := os.Stat(path.Dir(fileName))
+	if err != nil {
+		return 0, 0, err
+	}
+	dirSys := dirInfo.Sys()
+	var dirInode uint64
+	if dirStat, ok := dirSys.(*syscall.Stat_t); ok {
+		dirInode = uint64(dirStat.Ino)
+	}
+	fmt.Printf("Inode of Directory: %d\n", dirInode)
+
+	return inode, dirInode, nil
+}
+
```

Comment:
nit: I think implementing a helper to provide the inode of a file will be very useful here (and in other places as well). If you are for it, it'll great to implement this helper on `pkg/utils/inode`.

Reply:
yeah done, let me know if we need anything else tweaked!